.github/workflows/sign-and-release-to-github.yml

```diff
@@ -2,15 +2,56 @@ name: continuous prerelease to GitHub
 
 on:
   push:
-    branches: [ main ]
-
+    branches: [ "*" ]
   workflow_dispatch:
 
 jobs:
-  sign-tag-release:
+  prepare-deployment:
     runs-on: ubuntu-latest
+    outputs:
+      tag-slug: ${{ steps.determine-npm-tag.outputs.tag-slug }}
+      deployment-id: ${{ fromJson(steps.create-deployment.outputs.result).data.id }}
+    steps:
+      - name: Create GitHub Deployment
+        id: create-deployment
+        uses: actions/github-script@v5
+        with:
+          script: |
+            return await github.rest.repos.createDeployment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: context.sha,
+              environment: "review",
+              transient_environment: true,
+              auto_merge: false,
+              mediaType: {previews: ["flash", "ant-man"]},
+              // The deployment runs in parallel with CI, so status checks will never have succeeded yet:
+              required_contexts: [],
+            });
+
 
+  sign-tag-release:
+    runs-on: ubuntu-latest
+    needs: [prepare-deployment]
     steps:
+      - name: Mark GitHub Deployment as in progress
+        id: start-deployment
+        uses: actions/github-script@v5
+        env:
+          DEPLOYMENT_ID: "${{ needs.prepare-deployment.outputs.deployment-id }}"
+        with:
+          script: |
+            return await github.rest.repos.createDeploymentStatus({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              deployment_id: process.env.DEPLOYMENT_ID,
+              description: "Building add-on",
+              environment: "review",
+              log_url: `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`,
+              state: "in_progress",
+              mediaType: {previews: ["flash", "ant-man"]},
```

Comment:
What are these media types? I didn't find them anywhere in the docs.

Reply:
I think they were listed in the docs back when I first implemented this; they're used for API previews. Specifically, they're for fairly minor features we're using: [automatically labelling a Deployment as inactive](https://developer.github.com/changes/2016-04-06-deployment-and-deployment-status-enhancements/#automatic-creation-of-inactive-statuses) and [being able to list a Deployment as "in progress"](https://developer.github.com/changes/2018-10-16-deployments-environments-states-and-auto-inactive-updates/). I can't find any posts about those API's having graduated, but it also seems that the deployment still works just fine (including marking as "in progress") without them (d57000d), and they're no longer listed in the documentation examples, so I'll just keep them out.